akumulid/tcp_server.cpp

```diff
@@ -361,7 +361,13 @@ struct TcpServerBuilder {
 
     std::shared_ptr<Server> operator () (std::shared_ptr<DbConnection> con,
                                          std::shared_ptr<ReadOperationBuilder>,
-                                         const ServerSettings& settings) {
+                                         ServerSettings& settings) {
+        if ((int)std::thread::hardware_concurrency() <= 4) {
+            settings.nworkers = 1;
```

Comment:
You can use local variable instead of making settings object immutable:

```
auto nworkers = settings.nworkers;
```

and than pass this local variable to the constructor.